### ISSUE TITLE
Rearrange the order of the stats

### ISSUE THREAD

**Issue by maesa** [iglance/iGlance#23]
**Is your feature request related to a problem? Please describe.**
Currently, rearrange the order of the stats is done by tick/untick available stats.

**Describe the solution you'd like**
Movable/dragable stats.

**Comment by D0miH** [comment]
Yeah this is a good idea. 
However because there is apparently no easy way to get the position of the status bar items this might be a bit tricky. But I am sure we can think of a workaround.

**Comment by maesa** [comment]
Sorry, what I mean is movable/dragable in settings, not status bar.

![image](https://user-images.githubusercontent.com/7199717/53536785-3c75b600-3b3a-11e9-8d89-6c5b33aa9ee7.png)
or
![image](https://user-images.githubusercontent.com/7199717/53536827-64651980-3b3a-11e9-8895-430a61da9904.png)

**Comment by D0miH** [comment]
Ahh, ok! Thanks for the clarification. 👍 

So any suggestion on how we can integrate this into the current user interface?
Besides integrating this new feature, I was thinking about a redesign of the user interface, since the GUI seems pretty crammed up with all the lately added features. This might be a good opportunity. 
What do you guys think?

**Comment by Moneypulation** [comment]
@D0miH That's a good idea. Do you want to design the new GUI? I can implement the rearrangement feature once the GUI redesign is finished. (We'll probably need a new tab for the rearrangement feature. Maybe called "Other", where we can also move stuff like the update interval and the check update button.
Also what do you think about the idea to add a checkbox to toggle checking for updates automatically? By default this should be checked, but I'd also like to give the user the option to opt out

**Comment by D0miH** [comment]
@Moneypulation Yeah, i can do the GUI redesign. I will make a prototype and post an image of it here for feedback. 

A toggle for checking the updates is a good idea. This might prevent annoying popups when waking up from sleep mode.

**Comment by D0miH** [comment]
Here is my first prototype.
Feedback is welcome :)

![iglanceredesign](https://user-images.githubusercontent.com/36657497/53751067-b647cf80-3eab-11e9-8daf-f611b8db2c84.png)

**Comment by Moneypulation** [comment]
Visually it looks promising. Here are my points:

1. what use case does the uptime label provide in your opinion?
2. I assume you can click the labels on the left so that it opens the corresponding tab and simultaneously they can be rearranged through the up and down arrows. In this case we would need a distinction between cpu utilization and cpu temperature
3. I'd add a label "version x.y" at the bottom somewhere or maybe right below the logo
4. regarding the cpu and memory graph: we should keep in mind that iGlance stays lightweight and doesn't comsume much battery power. But it would also look good if the whole graph was rendered immediately once the user opens the GUI. A solution to this could be to save the values of something like the last 60 seconds in an array, e.g. an array of size 60 that constantly overwrites itself, and use those values to print the whole graph once the GUI is opened (but not in background) similar to how down and upload of the last hour is being calculated

**Comment by D0miH** [comment]
The uptime label is just some additional "nice to know" info. And the empty space on the dashboard would have looked a bit awkward. 
Yeah point 2 and 3 sounds reasonable and i can integrate that.
My idea for the graph was to calculate the average for every minute by saving the values in an array and then save the average values of every minute again in an array. This way we have a data point for the graph for every minute which allows us to display a longer history of the cpu usage. Using this method we could visualize the usage for 5 hours with an array of length 300. When the GUI is closed only the values are saved which shouldn't draw much power. When the GUI is opened the graphs are drawn using the saved values.

Because GUI redesign is a feature on its own I opened a new issue for that.

**Comment by D0miH** [comment]
I will post a new version of the prototype in #25.

**Comment by gihnius** [comment]
1. Move the Fan speed to the CPU or the Temperature menu
2. Show primary interface ip address, as well as public ip address(optional).
